src/lib/authenticate-cloud.ts

```diff
@@ -0,0 +1,39 @@
+import axios, { AxiosPromise, AxiosRequestConfig, AxiosResponse } from 'axios';
+
+interface AuthenticateCloudParams {
+  AM_URL: string;
+  realm: string;
+  username: string;
+  password: string;
+}
+interface CloudAuth {
+  tokenId: string;
+  successUrl: string;
+  realm: string;
+}
+async function authenticateCloud({
+  realm,
+  AM_URL,
+  username,
+  password,
+}: AuthenticateCloudParams): Promise<AxiosResponse<CloudAuth>> {
+  try {
+    const config: AxiosRequestConfig = {
+      baseURL: AM_URL.slice(-2),
```

Comment:
You just want the last two characters of the `AM_URL` string? So, given `https://auth.example.com/am` the `baseUrl` would be `am` and then `/open/am/json/realms/root/realms/${realm}/authenticate` would be appended?